### PR TITLE
Balancer

### PR DESCRIPTION
**Pull request by sevenshi**
Refactoring balancer    add  beets

### CODE REVIEW

File: subgraphs/balancer-v2/config/arbitrum.json [hunk 1]
```diff
@@ -11,5 +11,9 @@
     "wbtc": "0x2f2a2543B76A4166549F7aaB2e75Bef0aefC5B0f",
     "wmatic": "0x0000000000000000000000000000000000000000",
     "bal": "0x040d1EdC9569d4Bab2D15287Dc5A4F10F56a56B8"
-  }
+  },
+  "protocolName": "balancer V2",
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/config/arbitrum.json:15]: `Balancer v2`

File: subgraphs/balancer-v2/config/arbitrum.json [hunk 2]
```diff
@@ -11,5 +11,9 @@
...
+  "protocolSlug": "balancer V2",
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/config/arbitrum.json:16]: `balancer-v2`

File: subgraphs/balancer-v2/config/beets.json [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    "weth": "0x74b23882a30290451A17c44f4F05243b6b58C76d",
+    "wbtc": "0x321162Cd933E2Be498Cd2267a90534A804051b11",
+    "wftm": "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83",
+    "beets": "0xF24Bcf4d1e507740041C9cFd2DddB29585aDCe1e"
+  },
+
+  "protocolName": "beethovenx",
+  "protocolSlug": "beethovenx",
+  "defaultNetwork": "FANTOM",
+  "isBeest": true
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/config/beets.json:19]: `isBeets`

File: subgraphs/balancer-v2/config/beets.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "usdc": "0x04068DA6C83AFCFA0e13ba15A6696662335D5B75",
+    "fusdt": "0x049d68029688eAbF473097a2fC38ef61633A3C7A",
+    "mim": "0x82f0B8B456c1A451378467398982d4834b6829c1",
+    "weth": "0x74b23882a30290451A17c44f4F05243b6b58C76d",
+    "wbtc": "0x321162Cd933E2Be498Cd2267a90534A804051b11",
+    "wftm": "0x21be370D5312f44cB42ce377BC9b8a0cEF1A4C83",
+    "beets": "0xF24Bcf4d1e507740041C9cFd2DddB29585aDCe1e"
+  },
+
+  "protocolName": "beethovenx",
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/config/beets.json:16]: `Beethoven X`

File: subgraphs/balancer-v2/config/beets.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  "protocolSlug": "beethovenx",
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/config/beets.json:17]: `beethoven-x`

File: subgraphs/balancer-v2/src/common/updateMetrics.ts [hunk 6]
```diff
@@ -0,0 +1,250 @@
+// import { log } from '@graphprotocol/graph-ts'
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/src/common/updateMetrics.ts:1]: why rename from `metrics.ts`?

Comment (sevenshi) [reply subgraphs/balancer-v2/src/common/updateMetrics.ts:1]: all fixed  and rename to metrics.ts

File: subgraphs/balancer-v2/src/common/constants.mustache [hunk 7]
```diff
@@ -63,33 +87,71 @@ export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000";
...
 
 /////////////////////////////
 ///// Protocol Specific /////
 /////////////////////////////
 
+export let DEFAULTNETWORK =  "{{defaultNetwork}}" 
+
 export let VAULT_ADDRESS = Address.fromString('{{vault}}')
-export let FEE_COLLECTOR_ADDRESS = Address.fromString('{{feeCollector}}')
+{{#isBeest}}
```
Comment (this-username-is-taken) [inline subgraphs/balancer-v2/src/common/constants.mustache:138]: isBeets

**Comment by this-username-is-taken** [comment]
are you handling reward token for beets?

**Comment by sevenshi** [comment]
> are you handling reward token for beets?

not yet

**Pull request merged**